administrator/components/com_associations/controllers/association.php

```diff
@@ -17,6 +17,40 @@
 class AssociationsControllerAssociation extends JControllerForm
 {
 	/**
+	 * Method to edit an existing record.
+	 *
+	 * @param   string  $key     The name of the primary key of the URL variable.
+	 * @param   string  $urlVar  The name of the URL variable if different from the primary key
+	 *                           (sometimes required to avoid router collisions).
+	 *
+	 * @return  boolean  True if access level check and checkout passes, false otherwise.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function edit($key = null, $urlVar = null)
+	{
+		$cp   = AssociationsHelper::getComponentProperties($this->input->get('component', '', 'string'));
+		$id   = $this->input->get('id', 0, 'int');
+		$user = JFactory::getUser();
+
+		$table = clone $cp->table;
+		$table->load($id);
+
+		$checkedOut = $table->{$cp->fields->checked_out} == 0 || $table->{$cp->fields->checked_out} == $user->id;
+
+		// Attempt to check-out the new record for editing and redirect.
+		if (!$checkedOut && !$cp->model->checkout($id))
+		{
+			$this->setRedirect(JRoute::_('index.php?option=com_associations&view=associations', false));
+			JError::raiseError( 100, JText::sprintf('JLIB_APPLICATION_ERROR_CHECKOUT_FAILED', $cp->model->getError()));
+		}
+		else
+		{
+			return parent::display();
+		}
+	}
```

Comment:
improve suggestion
```php
	/**
	 * Method to edit an existing record in side by side view.
	 *
	 * @param   string  $key     The name of the primary key of the URL variable.
	 * @param   string  $urlVar  The name of the URL variable if different from the primary key
	 *                           (sometimes required to avoid router collisions).
	 *
	 * @return  boolean  True if checkout passes, false otherwise.
	 *
	 * @since   __DEPLOY_VERSION__
	 */
	public function edit($key = NULL, $urlVar = NULL)
	{
		$cp = AssociationsHelper::getComponentProperties($this->input->get('component', '', 'string'));

		if (!is_null($cp->fields->checked_out))
		{
			$id    = $this->input->get('id', 0, 'int');
			$user  = JFactory::getUser();
			$table = clone $cp->table;
			$table->load($id);

			// Attempt to check-out the record for editing and redirect.
			if (!in_array($table->{$cp->fields->checked_out}, array($user->id, 0) && !$cp->model->checkout($id))
			{
				JFactory::getApplication()->enqueueMessage(JText::sprintf('JLIB_APPLICATION_ERROR_CHECKOUT_FAILED', $cp->model->getError()), 'error');
				$this->setRedirect(JRoute::_('index.php?option=com_associations&view=associations', false));

				return false;
			}
		}

		return parent::display();
	}
```